rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -125,17 +127,43 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
             self.len = b.getRemainingImages(self.loader._handle)//self.bs
         else:
             self.len = b.getRemainingImages(self.loader._handle)
+        self.num_batches = self.loader._external_source.n // self.bs if self.loader._external_source.n % self.bs == 0 else (self.loader._external_source.n // self.bs + 1)
```

Comment:
Can you explain how the value is set here?

Reply:
@swetha097 Please check this

Reply:
This is to calculate the total number of batches to process for the external source
_external_source is the source which is the object of the source class.
n is defined as the dataset_size.

So when the dataset size is divisible by batchsize, we have dataset_size / bs
else the num_batches would be after padding

Reply:
@shobana-mcw confirm if that makes sense